lib/shared/puppet_forge/v3/module_release.rb

```diff
@@ -0,0 +1,72 @@
+require 'shared/puppet_forge/connection'
+
+module PuppetForge
+  module V3
+    # Access metadata and downloads for a specific module release.
+    class ModuleRelease
+
+      include PuppetForge::Connection
+
+      # @!attribute [r] full_name
+      #   @return [Forge::ModuleName] The hyphen delimited name of this module
+      attr_reader :full_name
+
+      # @!attribute [r] version
+      #   @return [String] The version of this module
+      attr_reader :version
+
+      # @param full_name [String] The name of the module, will be converted to
+      #   a PuppetForge::ModuleName instance.
+      # @param version [String]
+      def initialize(full_name, version)
+        @full_name = PuppetForge::ModuleName.new(full_name)
+        @version   = version
+      end
+
+      # @return [Hash] The metadata for this module
+      def data
+        @data ||= conn.get(resource_url).body
+      end
+
+      # @return [String] The unique identifier for this module release.
+      def slug
+        "#{full_name}-#{version}"
+      end
+
+      # Download this module release to the specified path.
+      #
+      # @param path [Pathname]
+      # @return [void]
+      def download(path)
+        resp = conn.get(file_url)
+        path.open('wb') { |fh| fh.write(resp.body) }
+      end
```

Comment:
Are we planning on more specific error handling at some point? It seems like we probably shouldn't even write to the file unless it was a 2xx response. (```resp.success?```)